app/code/Magento/Backend/Model/Dashboard/Chart.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Backend\Model\Dashboard;
+
+use Magento\Backend\Helper\Dashboard\Data as DataHelper;
+use Magento\Backend\Helper\Dashboard\Order as OrderHelper;
+use Magento\Backend\Model\Dashboard\Chart\Date;
+use Magento\Framework\App\RequestInterface;
+
+/**
+ * Dashboard chart data retriever
+ */
+class Chart
+{
+    /**
+     * @var RequestInterface
+     */
+    private $request;
+
+    /**
+     * @var Date
+     */
+    private $dateRetriever;
+
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @var DataHelper
+     */
+    private $dataHelper;
+
+    /**
+     * Chart constructor.
+     * @param RequestInterface $request
+     * @param Date $dateRetriever
+     * @param OrderHelper $orderHelper
+     * @param DataHelper $dataHelper
+     */
+    public function __construct(
+        RequestInterface $request,
+        Date $dateRetriever,
+        OrderHelper $orderHelper,
+        DataHelper $dataHelper
+    ) {
+        $this->request = $request;
+        $this->dateRetriever = $dateRetriever;
+        $this->orderHelper = $orderHelper;
+        $this->dataHelper = $dataHelper;
+    }
+
+    /**
+     * Get chart data by period and chart type parameter
+     *
+     * @param string $period
+     * @param string $chartParam
+     *
+     * @return array
+     */
+    public function getByPeriod($period, $chartParam): array
+    {
+        $this->orderHelper->setParam('store', $this->request->getParam('store'));
```

Comment:
Use request in the model is bad practice. Request objects should be used just in the controller. The model is another layer and it shouldn't be depended to the HTTP request

Reply:
How would you like to pass it to the model?

Reply:
@ihor-sviziev yes

Reply:
Done